IS0401Test.py

```diff
@@ -1102,8 +1102,9 @@ def test_21(self, test):
             # Advertise a registry at pri 0 and allow the Node to do a basic registration
             self.zc.register_service(registry_info)
 
-        # Wait for n seconds after advertising the service for the first POST from a Node
+        # Wait for n seconds after advertising the service for the first POST and then DELETE from a Node
         self.primary_registry.wait_for_registration(DNS_SD_ADVERT_TIMEOUT)
+        self.primary_registry.wait_for_delete(DNS_SD_ADVERT_TIMEOUT)

```

Comment:
Not critical because in success case there's unlikely to be any wait at all but does this second timeout logically need to be `DNS_SD_ADVERT_TIMEOUT`? Would `API_PROCESSING_TIMEOUT` (maybe `+ HTTP_TIMEOUT`?) make more sense?

Reply:
I agree it probably doesn't need to be as long as `DNS_SD_ADVERT_TIMEOUT`, I'm just a bit unsure about slightly overloading the documented purpose of the other constants. It could simply be a smaller hard-coded value to avoid placing extra entries in the config? 2 seconds does seem like plenty to allow between receiving the 200 response and issuing the DELETE.

Reply:
I would think the `HEARTBEAT_INTERVAL` would be appropriate. Before performing the next heartbeat, a Node should have DELETE the existing entry.

Reply:
That's true... if the Node were implemented incorrectly and ignored the result of the POST, it would make a heartbeat POST next, so waiting for `HEARTBEAT_INTERVAL` + a small margin seems reasonable, even though a well-implemented Node would have sent the DELETE "immediately".

Reply:
Thanks both. I've updated the second wait to use `HEARTBEAT_INTERVAL + 1` as-per the heartbeat wait periods earlier in the suite,